source/agora/common/crypto/Schnorr.d

```diff
@@ -230,31 +241,59 @@ public Signature sign (T) (
 public bool verify (T) (const ref Point X, const ref Signature sig, auto ref T data)
     nothrow @nogc @trusted
 {
-    Sig s = Sig.fromBlob(sig);
-    // Compute the challenge and reduce the hash to a scalar
-    Scalar c = hashFull(Message!T(X, s.R, data));
-    // Compute `R + c*X`
-    Point RcX = s.R + (c * X);
-    /// Compute `s.G`
-    auto S = s.s.toPoint();
-    return S == RcX;
+    try {
+        Sig s = Sig.fromBlob(sig);
+        if (s.s.isValid()) {
+            /// Compute `s.G`
+            auto S = s.s.toPoint();
+            // Compute the challenge and reduce the hash to a scalar
+            Scalar c = hashFull(Message!T(X, s.R, data));
+            // Compute `R + c*X`
+            Point RcX = s.R + (c * X);
+            return S == RcX;
+        } else {
+            return false;
+        }
```

Comment:
Prefer to write code with early returns. So:

```D
if (notValid())
    return;

// else
```

It avoids indentation and having to read a lot of code before knowing what the else part does.

Reply:
Done